ecosystem/node-checker/src/configuration/types.rs

```diff
@@ -229,6 +239,47 @@ impl NodeAddress {
 
         AptosRestClient::from((client, self.get_api_url()))
     }
+
+    /// Gets the NodeAddress as a NetworkAddress. If the URL is a domain name,
+    /// it will automatically perform DNS resolution. This method returns an
+    /// error if `public_key` is None.
+    pub fn as_noise_network_address(&self) -> Result<NetworkAddress> {
+        // Confirm we have a public key. Technically we can build a NetworkAddress
+        // without one, but it's not useful for any of our needs without one.
+        let public_key = match self.public_key {
+            Some(public_key) => public_key,
+            None => bail!("Cannot convert NodeAddress to NetworkAddress without a public key"),
+        };
+
+        // Ensure we can get socket addrs from the URL. If the URL is a domain
+        // name, it will automatically perform DNS resolution.
+        let socket_addrs = self
+            .url
+            .socket_addrs(|| None)
+            .with_context(|| format!("Failed to get SocketAddrs from address {}", self.url))?;
+
+        // Ensure this results in exactly one SocketAddr.
+        if socket_addrs.is_empty() {
+            bail!(
+                "NodeAddress {} did not resolve to any SocketAddrs. If DNS, ensure domain name is valid",
+                self.url
+            );
+        }
+        if socket_addrs.len() > 1 {
+            bail!(
+                "NodeAddress {} unexpectedly resolved to multiple SocketAddrs: {:?}",
+                self.url,
+                socket_addrs,
+            );
+        }
```

Comment:
What's wrong with this?  If it's using a load balancer it could have more than one (even though it's the same node).  I would probably go through all the socket addresses and see if one of them succeeds or all of them succeed

Reply:
Everywhere else in NHC assumes just one address, though there's nothing at the framework level enforcing this. It might be weird for this to test multiple addresses while everything else tests just one. Perhaps I can just take the first one for now instead of making this an error.